msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/SignInEmailOTPTest.kt

```diff
@@ -0,0 +1,96 @@
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+package com.microsoft.identity.client.e2e.tests.network.nativeauth
+
+import com.microsoft.identity.client.e2e.utils.assertState
+import com.microsoft.identity.internal.testutils.nativeauth.ConfigType
+import com.microsoft.identity.internal.testutils.nativeauth.api.TemporaryEmailService
+import com.microsoft.identity.nativeauth.statemachine.errors.SignInError
+import com.microsoft.identity.nativeauth.statemachine.errors.SubmitCodeError
+import com.microsoft.identity.nativeauth.statemachine.results.SignInResult
+import com.microsoft.identity.nativeauth.statemachine.results.SignUpResult
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert
+import org.junit.Test
+
+class SignInEmailOTPTest : NativeAuthPublicClientApplicationAbstractTest() {
+
+    private val tempEmailApi = TemporaryEmailService()
+
+    override val configType = ConfigType.SIGN_IN_OTP
+
+    /**
+     * Use valid email and OTP to get token and sign in.
+     * (hero scenario 6, use case 2.2.1, Test case 30)
+     */
+    @Test
+    fun testSuccess() {
+        retryOperation {
+            runBlocking { // Running with runBlocking to avoid default 10 second execution timeout.
+                val user = config.email
+                val signInResult = application.signIn(user)
+                assertState<SignInResult.CodeRequired>(signInResult)
+                val otp = tempEmailApi.retrieveCodeFromInbox(user)
+                val submitCodeResult = (signInResult as SignInResult.CodeRequired).nextState.submitCode(otp)
+                assertState<SignInResult.Complete>(submitCodeResult)
+            }
+        }
+    }
+
+    /**
+     * Use invalid email address to receive a "user not found" error.
+     * (use case 2.2.2, Test case 31)
+     */
+    @Test
+    fun testErrorIsUserNotFound() = runTest {
+        val user = config.email
```

Comment:
Can we avoid reading the valid email from the configuration, why not just generate a random string for the invalid username

Reply:
what is the benefit of doing one over the other?

Reply:
Sorry I haven't phrased the comment correctly. I meant avoid reading the correct email and then making it invalid by appending the extra character. This feels a little weird. Just create a random string for the invalid username.